test/e2e/framework/ginkgowrapper.go

```diff
@@ -31,3 +37,114 @@ func AnnotatedLocation(annotation string) types.CodeLocation {
 	codeLocation = types.NewCustomCodeLocation(annotation + " | " + codeLocation.String())
 	return codeLocation
 }
+
+// It is a wrapper around ginkgo.It which supports [WithFeatureGate],
+// [WithFeature], [WithEnvironment], [WithConformance], [WithNodeConformance]
+// as optional arguments in addition to those already supported by ginkgo
+// itself, like [ginkgo.Label].
+func It(text string, args ...interface{}) {
+	var ginkgoArgs []interface{}
+	var offset ginkgo.Offset
+
+	addLabel := func(label string) {
+		text += fmt.Sprintf(" [%s]", label)
+		ginkgoArgs = append(ginkgoArgs, ginkgo.Label(label))
+	}
+
+	for _, arg := range args {
+		switch arg := arg.(type) {
+		case withFeature:
+			addLabel("Feature:" + string(arg))
+		case withFeatureGate:
+			addLabel("FeatureGate:" + string(arg.featureGate))
+			addLabel(arg.level)
+		case withEnvironment:
+			addLabel("Environment:" + string(arg))
+		case withConformance:
+			addLabel("Conformance")
+		case withNodeConformance:
+			addLabel("NodeConformance")
+		case withDisruptive:
+			addLabel("Disruptive")
+		case withSerial:
+			addLabel("Serial")
+			ginkgoArgs = append(ginkgoArgs, ginkgo.Serial)
+		case ginkgo.Offset:
+			offset = arg
+		default:
+			ginkgoArgs = append(ginkgoArgs, arg)
+		}
+	}
+
+	offset++
+	ginkgoArgs = append(ginkgoArgs, offset)
+	ginkgo.It(text, ginkgoArgs...)
+}
+
+// TODO: Describe and Context wrappers.
+
+func WithFeature(name string) withFeature {
+	// TODO: verify that the feature is known.
```

Comment:
@aojea: do you agree that `WithFeature` and `WithEnvironment` should only accept "known" values?

We would have to create a registry similar to `FeatureGate` and add all values that are currently used in-tree to it by default.

Reply:
@SergeyKanzhelev how do you want to continue with https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/3041-node-conformance-and-features?